File: API/ImaloEducationApi/ImaloEducationApi/Models/AuditLogEntry.cs
namespace ImaloEducationApi.Models;

public sealed record AuditLogEntry
{
    public required int ScholarAuditLogId { get; init; }

    public required Guid ScholarId { get; init; }

    public required AuditAction ActionType { get; init; }

    public string? Details { get; init; }

    public required DateTime OccurredAt { get; init; }
}

public sealed record GlobalAuditLogEntry
{
    public required int ScholarAuditLogId { get; init; }

    public required Guid ScholarId { get; init; }

    public string? ScholarFirstName { get; init; }

    public string? ScholarLastName { get; init; }

    public required AuditAction ActionType { get; init; }

    public string? Details { get; init; }

    public required DateTime OccurredAt { get; init; }
}

File: UI/src/app/interfaces/audit-log-entry.ts
import { IsoDateTime } from './iso-date';

export type AuditAction = 'Created' | 'Edited' | 'Deleted';

export interface AuditLogEntry {
  scholarAuditLogId: number;
  scholarId: string;
  actionType: AuditAction;
  details: string | null;
  occurredAt: IsoDateTime;
}
